processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -450,7 +455,8 @@ public boolean isAssignableTo(Type other) {
      * @param cmStrategy collection mapping strategy
      * @return an unmodifiable map of all write accessors indexed by property name
      */
-    public Map<String, Accessor> getPropertyWriteAccessors( CollectionMappingStrategyPrism cmStrategy ) {
+    public Map<String, Accessor> getPropertyWriteAccessors( CollectionMappingStrategyPrism cmStrategy,
```

Comment:
This will only work when the `source` and `target` properties are the same.

If you modify the name in either `Source` or `Target` and accordingly use `@Mapping` then it'll fail.